k2/csrc/fsa_utils.h

```diff
@@ -111,6 +111,18 @@ std::string FsaToString(const Fsa &fsa, bool openfst = false,
                     order of FSAs in `src` must not be changed by this: i.e., if
                     we get the old fsa-index for each element of `order`, they
                     must be non-decreasing.
+                    Noted if the dest_state of an arc in the original Fsa does
+                    not exist in the output Fsa (i.e. the dest_state is not in
+                    `order`), then the arc will be removed in the returned
+                    FsaVec.
+                    For each Fsa to be renumbered, if the corresponding number
+                    of states in `order` for that Fsa is not zero, then it must
+                    be at least 2 (one of is the final state of that FSA);
+                    Otherwise, the algorithms will generate invalid FSA, i.e.
+                    the generated Fsa would have only one state. Noted the
+                    algorithm never check this, the caller should make sure
+                    that it must be at least of 2 (if not zero) and one of
```

Comment:
Haowen, this isn't the desired behavior, this algorithm should make sure never to generate an FSA with exactly one state.  We can detect an error and return  the empty FSA if the start state has arcs entering it that are not from the start state.